modules/misc.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Copyright 2022 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+function merge_mender_configs() {
+    declare -a ol_targets=()
+    local log="work/mergefiles/log"
+    local tmp="work/mergefiles/tmp"
+
+    for overlay in "${overlays[@]}"; do
+        [ -f "${overlay}/etc/mender/mender.conf" ] && ol_targets+=("${overlay}")
+    done
+
+    mkdir -p work/mergefiles
+    local -r menderconfs="input/resources/mender.conf ${ol_targets[@]/%//etc/mender/mender.conf}"
+
+    if [ "${#ol_targets[@]}" -ne 0 ]; then
+        echo "Attempting to merge /etc/mender/mender.conf in overlays. With respect to the overlays, the merge is done in the following order (first used as base):" >> "${log}"
```

Comment:
Suggestion: 

Use the log-primitives already present

Reply:
see the full log in the PRs header comment. the first is used as a base and then the rest are applied